tests/expected/zst/main.rs

```diff
@@ -0,0 +1,15 @@
+#[repr(C)]
+#[derive(Copy, Clone)]
+struct Z(i8, i64);
+
+struct Y;
+
+#[kani::proof]
+fn test_z() -> Z {
+    let m = Y;
+    let n = Y;
+    let zz = Z(1, -1);
+
+    let ptr: *const Z = if &n as *const _ == &m as *const _ { std::ptr::null() } else { &zz };
+    unsafe { *ptr }
```

Comment:
For Kani perhaps, but then it can no longer be exercised in Miri, concrete execution (including any of these in the Rust playground). So, no, I don't think this is a good idea until all questions around pointers and object lifetime have been settled by rust-lang.

Reply:
I personally prefer readability, since I can easily edit the test to run on Miri (and for concrete execution we can use concrete playback). In fact, you will need to at least remove the `#[kani::proof]` and change the function name to `main` to run in the playground.

That said, not a big deal and I understand your preference.